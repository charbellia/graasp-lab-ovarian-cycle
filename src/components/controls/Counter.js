import React from 'react';

export const Counter = ({ handleStart, handleStop }) => (
  <div className="Counter-Container">
    <div className="clock-container">
      <div className="clock-balls">
        <div className="clock-timer">
          <span className="clock-counter"> 01 </span>
          <span className="clock-dayer"> JOURS </span>
        </div>
      </div>
      <div className="clock-setter">
        <button className='button-start' onClick={handleStart}>START</button>
        <button className='button-stop' onClick={handleStop}>STOP</button>
      </div>
    </div>
  </div>
);

export default (Counter);
