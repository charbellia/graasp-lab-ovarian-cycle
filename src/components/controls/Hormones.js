import React from 'react';

export const Hormones = ({ d3 }) => (
  <div className="Hormone-Container">
    <div className="hormones-container">
      <p>
        <span className="fsh-puces">.</span>
        <span className="fsh-container">FSH</span>
        <span className="fsh-state">+</span>
      </p>
      <p>
        <span className="lh-puces">.</span>
        <span className="lh-container">LH</span>
        <span className="lh-state">+</span>
      </p>
      <p>
        <span className="oestro-puces">.</span>
        <span className="oestro-container">OESTROGÈNES</span>
        <span className="oestro-state">-</span>
      </p>
      <p>
        <span className="progest-puces">.</span>
        <span className="progest-container">PROGESTÉRONES</span>
        <span className="progest-state">-</span>
      </p>
    </div>
  </div>
);

export default (Hormones);
