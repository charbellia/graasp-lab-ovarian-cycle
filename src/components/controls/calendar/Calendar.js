import React from 'react';
import PropTypes from 'prop-types';
// creating an svg file and pass
// the ovarian cycle dayCount to display
const Calendar = ({ dayCount, themeColor }) => (
  <svg
    aria-hidden="true"
    data-prefix="far"
    data-con="calendar"
    className="svg-inline--fa fa-calendar fa-w-14"
    role="img"
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 448 512"
  >
    <g id="Page-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g id="calendar" fill={themeColor}>
        <g id="calendar-regular" fillRule="nonzero">
          <path
            d="M400,64 L352,64 L352,12 C352,5.4 346.6,0 340,0 L300,0 C293.4,0 288,5.4 288,12 L288,64 L160,64 L160,12 C160,5.4 154.6,0 148,0 L108,0 C101.4,0 96,5.4 96,12 L96,64 L48,64 C21.5,64 0,85.5 0,112 L0,464 C0,490.5 21.5,512 48,512 L400,512 C426.5,512 448,490.5 448,464 L448,112 C448,85.5 426.5,64 400,64 Z M394,464 L54,464 C50.7,464 48,461.3 48,458 L48,160 L400,160 L400,458 C400,461.3 397.3,464 394,464 Z"
            id="Shape"
          />
        </g>
        <text fill="black" id="date" fontFamily="Helvetica" fontSize="160" fontWeight="normal">
          <tspan x="168" y="356">{dayCount}</tspan>
        </text>
      </g>
    </g>
  </svg>
);

Calendar.propTypes = {
  dayCount: PropTypes.number.isRequired,
  themeColor: PropTypes.string.isRequired,
};

export default (Calendar);
