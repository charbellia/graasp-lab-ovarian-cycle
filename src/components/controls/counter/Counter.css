.clock-starter {
  text-align: center;
  padding: 10px;
}
.fa-calendar {
  width: 15rem;
  margin-bottom: 2rem;
  margin-top: 2rem;
}
.btn-lg {
  font-size: 2rem;
}
