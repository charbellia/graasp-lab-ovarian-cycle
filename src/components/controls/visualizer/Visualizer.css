.visualizer-container {
  background-color: #ffffff;
  border-left: 1px solid rgba(0, 0, 0, 0.12);
  height: 100%;
}
.fa-redo-alt {
  width: 2rem;
}
.observe-button, .test-button {
  width: 50%;
  padding-top: 1rem;
  padding-bottom: 1rem;
  border-radius: 0;
  font-size: 1.6rem;
  color: #000;
}
.test-button, .test-button:hover {
  border: 1px solid #dedede;
  color: #dedede;
  background-color: rgba(0,0,0,0);
}
.test-button-active, .observe-button-active {
  background-color: #dedede;
  font-weight: bold;
}
.test-button-active:hover,
.observe-button-active:hover {
  background-color: #EEEEEE;
  color: #555555;
}
#disabledBtn {
  cursor: help;
}
.btn, .modal-container button {
  outline: 0;
}
.modal-color-title, .modal-lang-title {
  font-size: 1.1rem;
}
.modal-lang-title {
  line-height: 2;
}
.modal-lang-container {
  margin-top: 1rem;
}
.modal-container .btn-secondary {
  position: fixed;
  bottom: 8%;
  right: 2%;
  background: #fafafa;
  border-color: #fafafa;
  border-radius: 50%;
  font-size: 1.8rem;
}
.modal-container .btn-secondary:active {
  background-color: #fafafa !important;
  border-color: #fafafa !important;
}
.modal-container .btn-secondary svg {
  margin-bottom: .3rem;
}
