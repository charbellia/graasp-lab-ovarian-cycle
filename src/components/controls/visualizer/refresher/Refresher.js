import React from 'react';
import Qs from 'qs';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Setting from './Setting';

const Refresher = ({
  reloadPage,
  t,
  onOpenModal,
  openModal,
  onCloseModal,
  handleChangeComplete,
  handleLang,
  themeColor,
  toggleTitle,
}) => {
  const {
    mode = 'default',
  } = Qs.parse(window.location.search, { ignoreQueryPrefix: true });
  return (
    <div className="modal-container">
      { mode === 'default' ? (
        <Setting
          onOpenModal={onOpenModal}
          onCloseModal={onCloseModal}
          openModal={openModal}
          handleChangeComplete={handleChangeComplete}
          handleLang={handleLang}
          themeColor={themeColor}
          toggleTitle={toggleTitle}
          reloadPage={reloadPage}
          t={t}
        />
      ) : ''
      }
    </div>
  );
};


Refresher.propTypes = {
  reloadPage: PropTypes.func.isRequired,
  onOpenModal: PropTypes.func.isRequired,
  onCloseModal: PropTypes.func.isRequired,
  handleChangeComplete: PropTypes.func.isRequired,
  handleLang: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  openModal: PropTypes.bool.isRequired,
  themeColor: PropTypes.string.isRequired,
  toggleTitle: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  themeColor: state.setting.themeColor,
});

export default connect(mapStateToProps)(Refresher);
