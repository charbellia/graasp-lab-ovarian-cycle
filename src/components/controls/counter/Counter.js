import React from 'react';
import PropTypes from 'prop-types';
import './Counter.css';
import { Button } from 'reactstrap';

const Counter = ({
  handleStart,
  handleStop,
  isStarted,
  ovulationActive,
  preOvulationActive,
  postOvulationActive,
  t,
}) => {
  const isDisabled = preOvulationActive || ovulationActive || postOvulationActive;
  // show stop button if simulation started otherwise show start
  const startButton = (isStarted)
    ? <Button onClick={handleStop} size="lg" disabled={isDisabled}>{t('Stop Cycle')}</Button>
    : <Button onClick={handleStart} size="lg" disabled={isDisabled}>{t('Launch Cycle')}</Button>;
  return (
    <div className="counter-container">
      <div className="clock-setter">
        {startButton}
        <p className="select-step">
          <span className="or">{t('OR')}</span>
        </p>
      </div>
    </div>
  );
};

Counter.propTypes = {
  handleStart: PropTypes.func.isRequired,
  handleStop: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  isStarted: PropTypes.bool.isRequired,
  ovulationActive: PropTypes.bool.isRequired,
  postOvulationActive: PropTypes.bool.isRequired,
  preOvulationActive: PropTypes.bool.isRequired,
};
export default (Counter);
