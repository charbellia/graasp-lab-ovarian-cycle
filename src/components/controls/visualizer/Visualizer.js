import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import Counter from '../counter/Counter';
import Phases from '../phases/Phases';
import Refresher from './refresher/Refresher';
import { AppState } from '../../../config/AppState';
import './Visualizer.css';
import {
  defaultLang,
  themeColor,
  titleState,
} from '../../../actions';
// importing all different child components and
// pass them params they need
export class Visualizer extends Component {
  state = AppState;

  handleChangeComplete = (color) => {
    const newColor = color.hex;
    const {
      dispatchThemeColor,
    } = this.props;
    dispatchThemeColor({ newColor });
    this.postMessage({
      themeColor: newColor,
    });
  }

  handleLang = (lang) => {
    const newLang = lang.value;
    const {
      dispatchDefaultLanguage,
    } = this.props;
    dispatchDefaultLanguage({ newLang });
    this.postMessage({
      defaultLang: newLang,
    });
  }

  onOpenModal = () => {
    this.setState({
      openModal: true,
    });
    this.postMessage({
      open_setting_modal: true,
    });
  }

  onCloseModal = () => {
    this.setState({
      openModal: false,
    });
    this.postMessage({
      open_setting_modal: false,
    });
  }

  postMessage = (data) => {
    const message = JSON.stringify(data);
    console.log('message', message);
    if (document.postMessage) {
      document.postMessage(message, '*');
    } else if (window.postMessage) {
      window.postMessage(message, '*');
    } else {
      console.error('unable to find postMessage');
    }
  };

  toggleTitle = () => {
    const { showTitle } = this.state;
    this.setState({ showTitle: !showTitle });
    const { dispatchTitleState } = this.props;
    dispatchTitleState({ showTitle });
    this.postMessage({
      show_title: showTitle,
    });
  }

  render() {
    const { openModal } = this.state;
    const {
      handleOvulation,
      handlePostOvulation,
      handlePreOvulation,
      reloadPage,
      handleStart,
      handleStop,
      isStarted,
      ovulationActive,
      postOvulationActive,
      t,
    } = this.props;
    return (
      <div className="visualizer-container">
        <Counter
          handleStart={handleStart}
          handleStop={handleStop}
          isStarted={isStarted}
          ovulationActive={ovulationActive}
          postOvulationActive={postOvulationActive}
          themeColor={themeColor}
          t={t}
        />
        <Phases
          handleOvulation={handleOvulation}
          handlePostOvulation={handlePostOvulation}
          handlePreOvulation={handlePreOvulation}
          t={t}
        />
        <ToastContainer autoClose={20000} />
        <Refresher
          reloadPage={reloadPage}
          onOpenModal={this.onOpenModal}
          onCloseModal={this.onCloseModal}
          openModal={openModal}
          handleChangeComplete={this.handleChangeComplete}
          handleLang={this.handleLang}
          toggleTitle={this.toggleTitle}
          t={t}
        />
      </div>
    );
  }
}

Visualizer.propTypes = {
  handleStart: PropTypes.func.isRequired,
  handleStop: PropTypes.func.isRequired,
  handleOvulation: PropTypes.func.isRequired,
  handlePostOvulation: PropTypes.func.isRequired,
  handlePreOvulation: PropTypes.func.isRequired,
  dispatchThemeColor: PropTypes.func.isRequired,
  dispatchDefaultLanguage: PropTypes.func.isRequired,
  dispatchTitleState: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  reloadPage: PropTypes.func.isRequired,
  isStarted: PropTypes.bool.isRequired,
  ovulationActive: PropTypes.bool.isRequired,
  postOvulationActive: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  themeColor: state.setting.themeColor,
});

const mapDispatchToProps = {
  dispatchThemeColor: themeColor,
  dispatchDefaultLanguage: defaultLang,
  dispatchTitleState: titleState,
};

const connectedComponent = connect(mapStateToProps, mapDispatchToProps)(Visualizer);

export default (connectedComponent);
