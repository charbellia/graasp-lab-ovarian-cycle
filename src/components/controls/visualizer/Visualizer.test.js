import React from 'react';
import { shallow } from 'enzyme';
import Visualizer from './Visualizer';
import Calendar from '../calendar/Calendar';
import Counter from '../counter/Counter';
import Hormones from '../hormones/Hormones';
import Phases from '../phases/Phases';
import Refresher from './Refresher';

describe('<Visualizer /> component', () => {
  const props = {
    dayCount: 0,
    handleStart: jest.fn(),
    handleStop: jest.fn(),
    handleOvulation: jest.fn(),
    handlePostOvulation: jest.fn(),
    handlePreOvulation: jest.fn(),
    reloadPage: jest.fn(),
    isStarted: false,
    ovulation: false,
    postOvulation: false,
    preOvulation: false,
    ovulationActive: false,
    postOvulationActive: false,
    preOvulationActive: false,
    secreteLhFsh: false,
    secreteOestro: false,
    secreteProgest: false,
    t: jest.fn(),
  };

  const component = shallow(<Visualizer {...props} />);

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('has one <div /> with class visualizer-container', () => {
    expect(component.find('div.visualizer-container').length).toEqual(1);
  });
  it('shows one <Calendar /> component', () => {
    expect(component.find(Calendar).length).toEqual(1);
  });
  it('shows one <Counter /> component', () => {
    expect(component.find(Counter).length).toEqual(1);
  });
  it('has one <Hormones /> component', () => {
    expect(component.find(Hormones).length).toBe(1);
  });
  it('has one <Phases /> component', () => {
    expect(component.find(Phases).length).toBe(1);
  });
  it('has one <Refresher /> component', () => {
    expect(component.find(Refresher).length).toBe(1);
  });
});
