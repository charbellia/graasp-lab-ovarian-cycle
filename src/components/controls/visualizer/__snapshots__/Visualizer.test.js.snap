// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Visualizer /> component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Visualizer
    dayCount={0}
    handleOvulation={[MockFunction]}
    handlePostOvulation={[MockFunction]}
    handlePreOvulation={[MockFunction]}
    handleStart={[MockFunction]}
    handleStop={[MockFunction]}
    isStarted={false}
    ovulation={false}
    ovulationActive={false}
    postOvulation={false}
    postOvulationActive={false}
    preOvulation={false}
    preOvulationActive={false}
    reloadPage={[MockFunction]}
    secreteLhFsh={false}
    secreteOestro={false}
    secreteProgest={false}
    t={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Counter
          handleStart={[MockFunction]}
          handleStop={[MockFunction]}
          isStarted={false}
          ovulationActive={false}
          postOvulationActive={false}
          preOvulationActive={false}
          t={[MockFunction]}
        />,
        <Phases
          handleOvulation={[MockFunction]}
          handlePostOvulation={[MockFunction]}
          handlePreOvulation={[MockFunction]}
          ovulationActive={false}
          postOvulationActive={false}
          preOvulation={false}
          preOvulationActive={false}
          t={[MockFunction]}
        />,
        <Calendar
          dayCount={0}
        />,
        <Hormones
          ovulation={false}
          postOvulation={false}
          preOvulation={false}
          secreteLhFsh={false}
          secreteOestro={false}
          secreteProgest={false}
          t={[MockFunction]}
        />,
        <Refresher
          reloadPage={[MockFunction]}
        />,
      ],
      "className": "visualizer-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleStart": [MockFunction],
          "handleStop": [MockFunction],
          "isStarted": false,
          "ovulationActive": false,
          "postOvulationActive": false,
          "preOvulationActive": false,
          "t": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleOvulation": [MockFunction],
          "handlePostOvulation": [MockFunction],
          "handlePreOvulation": [MockFunction],
          "ovulationActive": false,
          "postOvulationActive": false,
          "preOvulation": false,
          "preOvulationActive": false,
          "t": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dayCount": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "ovulation": false,
          "postOvulation": false,
          "preOvulation": false,
          "secreteLhFsh": false,
          "secreteOestro": false,
          "secreteProgest": false,
          "t": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "reloadPage": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Counter
            handleStart={[MockFunction]}
            handleStop={[MockFunction]}
            isStarted={false}
            ovulationActive={false}
            postOvulationActive={false}
            preOvulationActive={false}
            t={[MockFunction]}
          />,
          <Phases
            handleOvulation={[MockFunction]}
            handlePostOvulation={[MockFunction]}
            handlePreOvulation={[MockFunction]}
            ovulationActive={false}
            postOvulationActive={false}
            preOvulation={false}
            preOvulationActive={false}
            t={[MockFunction]}
          />,
          <Calendar
            dayCount={0}
          />,
          <Hormones
            ovulation={false}
            postOvulation={false}
            preOvulation={false}
            secreteLhFsh={false}
            secreteOestro={false}
            secreteProgest={false}
            t={[MockFunction]}
          />,
          <Refresher
            reloadPage={[MockFunction]}
          />,
        ],
        "className": "visualizer-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleStart": [MockFunction],
            "handleStop": [MockFunction],
            "isStarted": false,
            "ovulationActive": false,
            "postOvulationActive": false,
            "preOvulationActive": false,
            "t": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleOvulation": [MockFunction],
            "handlePostOvulation": [MockFunction],
            "handlePreOvulation": [MockFunction],
            "ovulationActive": false,
            "postOvulationActive": false,
            "preOvulation": false,
            "preOvulationActive": false,
            "t": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "dayCount": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ovulation": false,
            "postOvulation": false,
            "preOvulation": false,
            "secreteLhFsh": false,
            "secreteOestro": false,
            "secreteProgest": false,
            "t": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "reloadPage": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Visualizer /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Visualizer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Counter />,
        <Phases />,
        <Calendar />,
        <Hormones />,
        <Refresher />,
      ],
      "className": "visualizer-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleStart": undefined,
          "handleStop": undefined,
          "isStarted": undefined,
          "ovulationActive": undefined,
          "postOvulationActive": undefined,
          "preOvulationActive": undefined,
          "t": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleOvulation": undefined,
          "handlePostOvulation": undefined,
          "handlePreOvulation": undefined,
          "ovulationActive": undefined,
          "postOvulationActive": undefined,
          "preOvulation": undefined,
          "preOvulationActive": undefined,
          "t": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "dayCount": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "ovulation": undefined,
          "postOvulation": undefined,
          "preOvulation": undefined,
          "secreteLhFsh": undefined,
          "secreteOestro": undefined,
          "secreteProgest": undefined,
          "t": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "reloadPage": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Counter />,
          <Phases />,
          <Calendar />,
          <Hormones />,
          <Refresher />,
        ],
        "className": "visualizer-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleStart": undefined,
            "handleStop": undefined,
            "isStarted": undefined,
            "ovulationActive": undefined,
            "postOvulationActive": undefined,
            "preOvulationActive": undefined,
            "t": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleOvulation": undefined,
            "handlePostOvulation": undefined,
            "handlePreOvulation": undefined,
            "ovulationActive": undefined,
            "postOvulationActive": undefined,
            "preOvulation": undefined,
            "preOvulationActive": undefined,
            "t": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "dayCount": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ovulation": undefined,
            "postOvulation": undefined,
            "preOvulation": undefined,
            "secreteLhFsh": undefined,
            "secreteOestro": undefined,
            "secreteProgest": undefined,
            "t": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "reloadPage": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
